src/librustc_borrowck/borrowck/mir/elaborate_drops.rs

```diff
@@ -586,6 +586,11 @@ impl<'b, 'tcx> ElaborateDropsCtxt<'b, 'tcx> {
                             // drop elaboration should handle that by itself
                             continue
                         }
+                        TerminatorKind::Resume => {
+                            // We can replace resumes with gotos
+                            // jumping to a canonical resume.
```

Comment:
I'm still not very happy with the story here. I think I will probably revise the comment to try to reflect my lack of unsureness about how we got to this point.